candi/tools/update_kubernetes_patchversions.sh

```diff
@@ -28,36 +30,55 @@ function check_requirements() {
+  if yq -e e "select(.k8s.\"${1}\".controlPlane | has(\"kubeProxy\"))" ../version_map.yml >/dev/null 2>/dev/null; then
     NEW_DIGEST="$(crane digest "registry.k8s.io/kube-proxy:v${1}.${2}")"
-    yq -i e ".k8s.\"${1}\".controlPlane.kubeProxy = \"${NEW_DIGEST}\"" /deckhouse/candi/version_map.yml
+    yq -i e ".k8s.\"${1}\".controlPlane.kubeProxy = \"${NEW_DIGEST}\"" ../version_map.yml
   fi
   # Kube-scheduler
   NEW_DIGEST=""
-  if yq -e e "select(.k8s.\"${1}\".controlPlane | has(\"kubeScheduler\"))" /deckhouse/candi/version_map.yml >/dev/null 2>/dev/null; then
+  if yq -e e "select(.k8s.\"${1}\".controlPlane | has(\"kubeScheduler\"))" ../version_map.yml >/dev/null 2>/dev/null; then
     NEW_DIGEST="$(crane digest "registry.k8s.io/kube-scheduler:v${1}.${2}")"
-    yq -i e ".k8s.\"${1}\".controlPlane.kubeScheduler = \"${NEW_DIGEST}\"" /deckhouse/candi/version_map.yml
+    yq -i e ".k8s.\"${1}\".controlPlane.kubeScheduler = \"${NEW_DIGEST}\"" ../version_map.yml
   fi
 }
 
 check_requirements
 
-TMPDIR="$(mktemp -du)"
-mkdir -p "${TMPDIR}"
-cd "${TMPDIR}"
-
-for VERSION in $(yq e /deckhouse/candi/version_map.yml -j | jq -r '.k8s | keys[]'); do
-  for PATCH in $(yq e /deckhouse/candi/version_map.yml -j | jq -r --arg version "${VERSION}" '.k8s."\($version)".patch'); do
+for VERSION in $(yq e ../version_map.yml -o json | jq -r '.k8s | keys[]'); do
+  for PATCH in $(yq e ../version_map.yml -o json | jq -r --arg version "${VERSION}" '.k8s."\($version)".patch'); do
     # Get last patch version from github CHANGELOG.md
     NEW_FULL_VERSION="$(curl -s "https://raw.githubusercontent.com/kubernetes/kubernetes/master/CHANGELOG/CHANGELOG-${VERSION}.md" | grep '## Downloads for v' | head -n 1 | grep -Eo "${VERSION}.[0-9]+")"
     NEW_PATCH="$(awk -F "." '{print $3}' <<< "${NEW_FULL_VERSION}")"
     if [[ "${NEW_PATCH}" -ne "${PATCH}" ]]; then
-      echo "New kubernetes patch version ${NEW_PATCH} found for ${VERSION}"
+      PR_DESCRIPTION="${PR_DESCRIPTION}$(echo -e "\n* New kubernetes patch version ${VERSION}.${NEW_PATCH}.")"
+      CREATE_PR=true
       update_version_map "${VERSION}" "${NEW_PATCH}"
     fi
   done
 done
 
-cd -
+if [[ "${CREATE_PR}" == "true" ]]; then
+  cd ../../
+  BRANCH="kubernetes-patchversions-$(date +"%y-%m-%d-%H-%M")"
+  git checkout -b "${BRANCH}"
+  git add .
+  git commit -m "[candi] New kubernetes control-plane components patchversions"
+  git push --set-upstream origin "${BRANCH}"
+  cat <<EOF | gh pr create -d -B main -F - -t "New kubernetes patchversions"
+## Description
+New Kubernetes control-plane components patchversions.
+${PR_DESCRIPTION}
+## Why do we need it, and what problem does it solve?
+Kubernetes control-plane components should be up to date.
+## Changelog entries
+\`\`\`changes
+section: candi
+type: feature
+summary: New Kubernetes control-plane components patchversions.
+impact: Restart Kubernetes control-plane components.
+impact_level: default
```

Comment:
I thought that impact should be `high` if control plane components are restarted. Otherwise, there will be no note about the impact in the generated changelog file.

Reply:
Is control plane restarts important to business-oriented users?

Reply:
Yes, and it is always mentioned in the release notes. Not only control plane, but all components.

Reply:
removed create of PR